src/include/storage/sql_table.h

```diff
@@ -125,6 +140,7 @@ class SqlTable {
                 ->GetTupleSlot() == slot,
         "This Delete is not the most recent entry in the txn's RedoBuffer. Was StageDelete called immediately before?");
 
+    // TODO(Schema-Change): among all relavant datatables.
```

Comment:
am I allowed to delete a tuple in a Schema with layout_version larger than `layout_version`?

Reply:
I should not be touching the tuple in the new version. Rather, I should look for the 'deleted' tuple in the old version by tracing the version chain to find the one visible to me.

Reply:
I should not be looking at the new version's DataTable. Rather, I should go to the table I can see (<= 'layout_version'), and find the tuple: 
- which can already be 'deleted' by another transaction that moves the data to a new schema. That transaction must be in-flight with me (otherwise I should see another version layout), and write-write conflict successfully caught. 
- which can still in my visible schema (might be write-locked, or available)